packages/actions/src/Concerns/CanImportRecords.php

```diff
@@ -80,7 +81,37 @@ protected function setUp(): void
                 ->label(__('filament-actions::import.modal.form.file.label'))
                 ->placeholder(__('filament-actions::import.modal.form.file.placeholder'))
                 ->acceptedFileTypes(['text/csv', 'text/x-csv', 'application/csv', 'application/x-csv', 'text/comma-separated-values', 'text/x-comma-separated-values', 'text/plain', 'application/vnd.ms-excel'])
-                ->rule('extensions:csv,txt')
+                ->rules([
+                    'extensions:csv,txt',
+                    File::types(['csv', 'txt'])->rules([
+                        function (string $attribute, mixed $value, Closure $fail) use ($action) {
+                            $csvStream = $this->getUploadedFileStream($value);
+                            $csvReader = CsvReader::createFromStream($csvStream);
+
+                            $csvReader->setHeaderOffset($action->getHeaderOffset() ?? 0);
+                            $csvColumns = $csvReader->getHeader();
+
+                            $duplicates = [];
+                            foreach (array_count_values($csvColumns) as $value => $count) {
+                                if ($count > 1) {
+                                    $duplicates[] = $value;
+                                }
+                            }
+
+                            if (! empty($duplicates)) {
+                                $columns = Arr::where($duplicates, fn (string $value): bool => $value !== '');
+
+                                if (empty($columns)) {
+                                    $fail(__('filament-actions::import.failure_csv.empty_columns'));
+                                } else {
+                                    $fail(__('filament-actions::import.failure_csv.duplicate_columns', [
+                                        'columns' => implode(', ', $columns),
+                                    ]));
+                                }
+                            }
```

Comment:
Can this be refactored to utilise early returns so that we don't have nested conditions?

Reply:
if(! empty($duplicates)) removed from nesting and returns